### PR TITLE
Sa/calibration_updates

### PR DESCRIPTION
**Pull request by sajjadazimi**
## Purpose 
Update the repo with the latest updates to the calibration tools

## Content
- Optimizing with averaged data over time and space (filtered data)
- UKI optimizations feature
- More post-processing tools

### CODE REVIEW

**Comment by sajjadazimi** [comment]
@trontrytel I'm sorry it's a lot of lines of codes again, but don't spend too much time looking at all changes. Maybe just skim through it. All the new things are in the calibration side of the package. New features are tested and work as expected. I promise to make smaller PRs in the future :))

File: Project.toml [hunk 1]
```diff
@@ -27,7 +27,6 @@ UnPack = "3a884ed6-31ef-47d7-9d2a-63182c4928ed"
 [compat]
 ClimaCore = "0.10"
 ClimaCorePlots = "0.2"
-CloudMicrophysics = "0.9"
```
Comment (trontrytel) [inline Project.toml:30]: Was that intentional?

File: test/Project.toml [hunk 2]
```diff
@@ -27,7 +27,6 @@ UnPack = "3a884ed6-31ef-47d7-9d2a-63182c4928ed"
 CLIMAParameters = "0.6"
 ClimaCore = "0.10"
 ClimaCorePlots = "0.2"
-CloudMicrophysics = "0.9"
```
Comment (trontrytel) [inline test/Project.toml:30]: Was that intentional?

**Review by trontrytel — APPROVED** [review]
I can't really review the calibration part, so I'm not sure I should be the reviewer here. Looks good to me, aside from the question about the changes in the toml files. 

Looks good to merge after squashing and rebasing things.

Maybe one point could be to finally add some things to our documentation of the KiD model. And give a small intro to the different calibration options you are introducing?

**Comment by trontrytel** [comment]
Also, thank you for adding unit tests and experiments that generate plots in the CI. This is super useful!

**Pull request merged**